src/plugins/data/public/query/query_string/dataset_service/types.ts

```diff
@@ -82,4 +96,14 @@ export interface DatasetTypeConfig {
    * Returns a list of sample queries for this dataset type
    */
   getSampleQueries?: (dataset: Dataset, language: string) => any;
+  /**
+   * Service used for indexedViews related operations
+   */
+  indexedViewsService?: DatasetIndexedViewsService;
```

Comment:
is this the right name? i don tthink the config should define the service. should it not be a service defined?